src/app/public/modules/lookup/types/lookup-add-click-event.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Specifies a callback function for the consumer to use to notify the lookup that a new item has been added.
+ */
+export interface SkyLookupAddMoreClickedEvent {
+
+  /**
+   * A callback function for the consumer to use to notify the lookup that a new item has been added.
+   * @param item The new item which has been added to the data. This item will be automatically selected.
+   * @param data The new state of the data source for the lookup component to search when users
+   * enter text. If not specified, the component will use the current state of the lookup
+   * component's `data` input; however, if this is not yet updated the new item will not be
+   * automatically selected.
+   */
+  itemAdded: (item: any, data?: any[]) => void;
```

Comment:
Instead of having `item` and `data` parameters, I'd prefer to see this be one `args` parameter with `item` and `data` properties. That way as we add new optional parameters, we can just add it to the args object instead of having a long list of optional parameters, which keeps consumers from having to do something like this:

```
eventArgs.itemAdded(myNewItem, myNewDataSource, undefined, undefined, undefined, theLastOptionalThingICareAbout);
```

Reply:
Done. I created a new class for this.